adds generic encoder/decoder for valueclasses.

pjan opened the pull request:
Attempt to address #469.

I'm not really sure how to add it to the `auto` package.
At the same time, in `semiauto`, the `deriveValueClassEncoder` works, but there's implicit resolution issues with the `deriveValueClassDecoder` that I seem unable to wrap my head around at this point.

Happy to make changes/amendments base upon input, @travisbrown.

Wogan commented on modules/generic-extras/src/main/scala/io/circe/generic/extras/decoding/ValueClassDecoder.scala:
```diff
@@ -0,0 +1,19 @@
+    implicit
+    gen:    Lazy[Generic.Aux[A, R :: HNil]],
+    decode: Decoder[R]
+  ): Decoder[A] = new ValueClassDecoder[A] {
```
Shouldn't this return type be ValueClassDecoder[A] not just Decoder[A] ?

pjan commented:
@travisbrown I fixed the PR, and added tests (currently for a ValueClass wrapping a String, but that should be sufficient). This is currently only for semiauto derivation.

If you want it for fully automatic derivation, please provide some pointers on how you envision it, and I can try to find some time.

fwiw, we keep repeating code for this in all our projects, so it would be great if this could make it in a release sooner rather than later. That's why I'm happy to put in some extra effort to get it done.
